earthly/rust/Earthfile

```diff
@@ -0,0 +1,75 @@
+# Common Rust UDCs and Builders.
+VERSION 0.7
+
+# cspell: words rustup automake autotools xutils miri nextest kani
+
+# Base Rustup build container.
+rustup:
+    # Base it on Debian bookworm-slim
+    FROM debian:bookworm-slim
+
+    WORKDIR /root
+
+    # Install necessary packages 
+    # Expand this list as needed, rather than adding more tools in later containers.
+    RUN apt-get update && \
+        apt-get install --no-install-recommends -y \
+            ca-certificates \
+            curl \
+            file \
+            build-essential \
+            autoconf \
+            automake \
+            autotools-dev \
+            libtool \
+            xutils-dev \
+            mold \
+            python3 \
+            python3-pip \
+            musl-tools && \
+        rm -rf /var/lib/apt/lists/*
+
+    # Install OpenSSL here but ONLY if we can't avoid using it.
+
+    # Install rustup.
+    RUN curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain none -y -v
+
+    # Set necessary env vars from this setup.
+    ENV PATH=/root/.cargo/bin:$PATH
+
+# Common Rust setup
+# Parameters:
+#  * toolchain : The `rust-toolchain` toml file.
+RUST_SETUP:
+    COMMAND
+    ARG toolchain
```

Comment:
You could probably set a default value here, since the file is typically always given the same name.

Reply:
Done